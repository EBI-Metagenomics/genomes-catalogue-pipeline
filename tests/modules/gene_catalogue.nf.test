nextflow_process {

    name "Test Process GENE_CATALOGUE"
    script "modules/gene_catalogue.nf"
    process "GENE_CATALOGUE"

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = file("$baseDir/tests/modules/fixtures/gene_catalogue/cluster_reps_ffn")
                input[1] = file("$baseDir/tests/modules/fixtures/gene_catalogue/mmseq_100")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
