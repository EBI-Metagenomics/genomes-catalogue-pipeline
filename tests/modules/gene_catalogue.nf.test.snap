{
    "Should run without failures": {
        "content": [
            {
                "gene_catalogue": [
                    [
                        "clusters.tsv:md5,c60d9b2487dfbd26a65513e51e143001",
                        "gene_catalogue-100.ffn:md5,880899eae713891a27dc10e9e5b99ee8"
                    ]
                ]
            }
        ],
        "timestamp": "2023-02-14T21:27:21+0000"
    }
}